File: 2 Regression/F Random Forest/my_random_forest_regression.r
# 76 Random Forest Regression in R

#copypasta
# Regression Template

# Importing the dataset
dataset = read.csv('Position_Salaries.csv')
dataset = dataset[2:3]

# Splitting the dataset into the Training set and Test set
# # install.packages('caTools')
# library(caTools)
# set.seed(123)
# split = sample.split(dataset$Salary, SplitRatio = 2/3)
# training_set = subset(dataset, split == TRUE)
# test_set = subset(dataset, split == FALSE)

# Feature Scaling
# training_set = scale(training_set)
# test_set = scale(test_set)

# Fitting the DT Model !!! --- !!! L 68
install.packages('randomForest')
library(randomForest)
set.seed(1234) # random seed generator
regressor = randomForest(x = dataset[1], #index in brackets gives column (dataframe) to use as ind var
                         y = dataset$Salary, # $qq gives qq as the vector to use for dep var
                         ntree = 100)


# Predicting a new result
y_pred = predict(regressor, data.frame(Level = 6.5))


# Visualising the Regression Model results (for higher resolution and smoother curve)
# install.packages('ggplot2')
library(ggplot2)
x_grid = seq(min(dataset$Level), max(dataset$Level), 0.01)
ggplot() +
  geom_point(aes(x = dataset$Level, y = dataset$Salary),
             colour = 'red') +
  geom_line(aes(x = x_grid, y = predict(regressor, newdata = data.frame(Level = x_grid))),
            colour = 'blue') +
  ggtitle('Truth or Bluff (Random Forest Model)') +
  xlab('Level') +
  ylab('Salary')

File: 2 Regression/F Random Forest/my_random_forest_regression.py
# Random Forest Regression

# Lecture 74 
#copypasta
# Regression Template

# Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importing the dataset
dataset = pd.read_csv('Position_Salaries.csv')
X = dataset.iloc[:, 1:2].values
y = dataset.iloc[:, 2].values

# Splitting the dataset into the Training set and Test set
"""from sklearn.cross_validation import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.2, random_state = 0)"""

# Feature Scaling
"""from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)
X_test = sc_X.transform(X_test)
sc_y = StandardScaler()
y_train = sc_y.fit_transform(y_train)"""

# Fitting the Forest Model !!! --- !!! start here
from sklearn.ensemble import RandomForestRegressor
regressor = RandomForestRegressor(n_estimators = 300, random_state = 0)
# can do more than 10 trees  /less than 300 in the forest (preferred)
# but do not expect # of steps to increase = to # of tree increase
# because steps represent average in a certain range/state
regressor.fit(X, y)
# gives output

# Predicting a new result
y_pred = regressor.predict(6.5)


# Visualising the Regression results (for higher resolution and smoother curve)
#higher resolution by changing val on first line (smaller = more res)
X_grid = np.arange(min(X), max(X), 0.001)
X_grid = X_grid.reshape((len(X_grid), 1))
plt.scatter(X, y, color = 'red')
plt.plot(X_grid, regressor.predict(X_grid), color = 'blue')
plt.title('Truth or Bluff (Random Forest Model)')
plt.xlabel('Position level')
plt.ylabel('Salary')
plt.show()